maven-core/src/main/java/org/apache/maven/internal/impl/DefaultArtifact.java

```diff
@@ -95,7 +97,12 @@ public String getClassifier() {
 
     @Override
     public boolean isSnapshot() {
-        return DefaultModelVersionParser.checkSnapshot(artifact.getVersion());
+        return artifact.isSnapshot();
```

Comment:
We had this discussion weeks ago, I think we agreed the snapshot knowledge was not on the artifact.  We now have a pluggable `VersionScheme`, I think it belongs there.

Anyway, I think those changes related to snapshots are unrelated and would be better in a separate PR to ease the discussion.

Reply:
Agreed. 

My point is to have one "source of truth". So if we change this once, we change that one spot. Currently, resolver Artifact.isSnapshot should be it IMHO. But agree, let's move it out from this PR, does not belong here.